src/variant/variant.cpp

```diff
@@ -303,123 +303,131 @@ Variant::operator float() const {
 }
 
 Variant::operator String() const {
-	String result;
-	to_type_constructor[STRING](result._native_ptr(), _native_ptr());
-	return result;
+	return String(this);
 }
 
 Variant::operator Vector2() const {
+	// @todo Avoid initializing result before calling constructor (which will initialize it again)
```

Comment:
Is it possible?

Reply:
It is possible, but we decided to postpone it to later.

This PR fixes the memory leak. Avoiding initializing those other results would be a slight performance improvement, but those aren't leaking memory, so fixing it isn't as urgent. The `@todo`'s are there to remind us to fix it eventually, though :-)